src/MIDebugEngine/AD7.Impl/AD7Engine.cs

```diff
@@ -590,10 +590,8 @@ public int Detach()
         {
             _breakpointManager.ClearBoundBreakpoints();
 
-            _pollThread.RunOperation(new Operation(delegate
-            {
-                _debuggedProcess.Detach();
-            }));
+            _pollThread.RunOperation(() => _debuggedProcess.CmdDetach());
+            _debuggedProcess.Detach();
```

Comment:
Can you remove the Detach method from _debuggedProcess while you are in here? Otherwise looks good to me.